### DISCUSSION TITLE
ignoreDifferences with ApplicationSets

### DISCUSSION THREAD

**Discussion by afergie-r7** [argoproj/argo-cd#10796] [Q&A]
Having an issue with checksums out of our control listing several applications as out of sync, reading the docs it seems as though the solution to this would be to use ignoreDifferences to exclude these specific fields, however as the apps themselves are of kind 'ApplicationSet' ignoreDifferences is invalid in them and as such the ignoreDifferences is added to the top level app of the ApplicationSet and it seems these are not being passed into the application 

Example 

![Screenshot 2022-10-05 at 11 07 22](https://user-images.githubusercontent.com/100145565/194036298-ea6f2f5e-c24d-4b5f-9501-4af5c3b6fec3.png)

![Screenshot 2022-10-05 at 11 09 18](https://user-images.githubusercontent.com/100145565/194036392-1a1362bb-0e22-43de-9bab-d0870b2a7ae6.png)

**Comment by jgwest** [comment]
Hi @afergie-r7, you should be able to specify ignore differences from within an `Application` template inside an ApplicationSet, unless I am missing something...

**Reply by TimJones** [reply]
That can be done, but then the `ignoreDifferences` is added to *all* Applications created by the ApplicationSet, which is sub-optimal.

**Comment by martinkocjan-tomtom** [comment]
> That can be done, but then the `ignoreDifferences` is added to _all_ Applications created by the ApplicationSet, which is sub-optimal.

You can use `templatePatch` and apply it only for the apps you want:
https://argo-cd.readthedocs.io/en/latest/operator-manual/applicationset/Template/#template-patch

**Reply by trc-ikeskin** [reply]
Could you kindly give a brief example on how that would work? Would you target the specific application through helm-if-clauses?

**Reply by AndreCbrera** [reply]
may can you use this:

https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Controlling-Resource-Modification/#ignore-certain-changes-to-applications

I think me answer is a little late but who knows
